aiida_nanotech_empa/workflows/gaussian/natorb_workchain.py

```diff
@@ -0,0 +1,184 @@
+from aiida_nanotech_empa.workflows.gaussian import common
+
+import numpy as np
+
+from aiida.engine import WorkChain, ToContext, calcfunction, ExitCode
+from aiida.orm import Code, Dict, RemoteData
+
+from aiida.plugins import WorkflowFactory
+
+GaussianBaseWorkChain = WorkflowFactory('gaussian.base')
+
+## --------------------------------------------------------------------
+## Natural orbital processing:
+
+
+def standard_num_odd(no_occs):
+    n_odd = 0.0
+    for n in no_occs:
+        n_odd += 2 * n - n**2
+    return n_odd
+
+
+def head_gordon_num_odd(no_occs):
+    n_odd = 0.0
+    for n in no_occs:
+        n_odd += 1 - np.abs(1 - n)
+    return n_odd
+
+
+def head_gordon_alt_num_odd(no_occs):
+    n_odd = 0.0
+    for n in no_occs:
+        n_odd += n**2 * (2 - n)**2
+    return n_odd
+
+
+def spin_proj_nakano(no_occs, i_hono=None):
```

Comment:
move to utils?

Reply:
Well this code is only used for the natorb workflow. It's convenient to have the implementation right there. Possibly if it's needed anywhere else (i doubt), the we could move it to utils.